From 6054fe3bc7e73a569d70bfb8a457f679a87406bc Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 11 Dec 2024 17:25:55 +0800
Subject: [PATCH 228/260] AOSCOS: staging: sb105x: revise type of second
 argument of mp_send_xchar() as u8

Per commit 3a00da027946 ("tty: make tty_operations::send_xchar accept u8
char"), the type of the second argument of member function
`mp_send_xchar()' in `struct tty_operations' was revised from type `char'
to type `u8'.

Revise driver code accordingly.

Fixes: "AOSCOS: Revert "staging: sb105x: delete the driver""
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/staging/sb105x/sb_pci_mp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/sb105x/sb_pci_mp.c b/drivers/staging/sb105x/sb_pci_mp.c
index 8712e7423064..19098bbfa113 100644
--- a/drivers/staging/sb105x/sb_pci_mp.c
+++ b/drivers/staging/sb105x/sb_pci_mp.c
@@ -43,7 +43,7 @@ static ssize_t mp_write(struct tty_struct *tty, const unsigned char *buf, long u
 static unsigned int mp_write_room(struct tty_struct *tty);
 static unsigned int mp_chars_in_buffer(struct tty_struct *tty);
 static void mp_flush_buffer(struct tty_struct *tty);
-static void mp_send_xchar(struct tty_struct *tty, char ch);
+static void mp_send_xchar(struct tty_struct *tty, u8 ch);
 static void mp_throttle(struct tty_struct *tty);
 static void mp_unthrottle(struct tty_struct *tty);
 static int mp_get_info(struct sb_uart_state *state, struct serial_struct *retinfo);
@@ -703,7 +703,7 @@ static void mp_flush_buffer(struct tty_struct *tty)
 	tty_wakeup(tty);
 }
 
-static void mp_send_xchar(struct tty_struct *tty, char ch)
+static void mp_send_xchar(struct tty_struct *tty, u8 ch)
 {
 	struct sb_uart_state *state = tty->driver_data;
 	struct sb_uart_port *port = state->port;
-- 
2.48.1

